test/shared-class.test.js

```diff
@@ -29,11 +29,6 @@ describe('SharedClass', function() {
       expect(sc.http.path).to.equal('/some');
     });
 
-    it('fills http.path using a normalized path', function() {
-      var sc = new SharedClass('SomeClass', SomeClass, {normalizeHttpPath: true});
```

Comment:
> As far as I can tell, so far, the app-level setting had no effect whatsoever, and only the model level was actually sort of working, but only one level deep: nested routes were reverting to the default of not normalizing the paths, ignoring the model-level setting (as well as the app-level one).

Yes, your assessment looks correct to me.

>  Is any app constructing `SharedClass` instances directly? 

Whenever you create a new model in your LoopBack application, a new `SharedClass` instance is created for this model. See [lib/model.js#L206-L211](https://github.com/strongloop/loopback/blob/ca3a21ddd5f4e29d4996858b126ca8b53858ff1c/lib/model.js#L206-L211) in strongloop/loopback.

Here is the use case we need to preserve:
 - create a LoopBack app
 - add a `FooBar` model
 - configure the model to `normalizeHttpPath` via `common/models/foobar.json`
 - observe that the model is mounted at `/api/foo-bar`

Such application should keep working (with `/api/foo-bar`) even after upgrading to your new version of strong-remoting, without any configuration changes needed. That's our understanding of backwards compatibility.

> I don't think it makes sense to transform the http routes on the SharedClass level, and that's what @ritch's original comment was about, I believe.

I am afraid I don't know what do you mean by @ritch's original comment, could you please paste a link? You can get the link by clicking on a date in the comment header.

To me, it makes a lot of sense to allow LoopBack users to configure normalized HTTP paths at multiple levels:

 - app-wide (as you are implementing here)
 - individually for each model (as is implemented now, with nested routes not normalized)